### PR TITLE
Fix for PHP7

### PR DESCRIPTION
**Pull request by aleextra**
Please see code update that fixes reassign error on mvc_controller.php at extract function on line #255

### CODE REVIEW

File: core/controllers/mvc_controller.php [hunk 1]
```diff
@@ -13,7 +13,8 @@ class MvcController {
     public $view_vars = array();
     
     function __construct() {
-    
+
+        session_start();
```
Comment (cyberscribe) [inline core/controllers/mvc_controller.php]: WordPress by design does not force a session_start(). Why would we want to break from that pattern?

Comment (cyberscribe) [inline core/controllers/mvc_controller.php]: See: https://www.quora.com/Why-does-WordPress-not-use-PHP-session-at-all/answer/Mark-Jaquith?srid=uVfU

**Comment by cyberscribe** [comment]
I don't see any code changes other than the session_start here, which is not something we would want to force a user to do. If there are still compatibility issues with PHP 7.2+ please submit another pull request as we would very much like to maintain forward-compatibility wherever possible. Thanks!

File: core/controllers/mvc_controller.php (hunk 1 above)
Comment (Surgeonito) [inline core/controllers/mvc_controller.php]: Well, yes.
the link you provided is explaining why WordPress uses its own login session mechanism. And it's clear.
But:
wp-mvc uses $_SESSION array only for flash() purpose - there is only $_SESSION['mvc_flash'] used within wp-mvc.
You want flash to  rely on $_SESSION and you do not want to make sure that the session is started. 
Without this 'start_session()' flash will fail sometimes.
So there are two paths:
- start the session as we did
or
- rebuild the 'flash' functionality
We added session_start()

**Comment by Surgeonito** [comment]
Hi

> I don't see any code changes other than the session_start here, 

please take a closer look at https://github.com/tombenner/wp-mvc/issues/166

> The problematic spot is:
> line 33, mvc_dispatcher.php:
> $controller->set('this', $controller);
You promised to contact with main developer and ask him about this very line.
This line is off now in our fork.

**Comment by cyberscribe** [comment]
Hi,

Thanks for explaining all this!

Regarding #166 - please can you make a separate pull request for this? I will again tag Tom Benner. It would be nice to know what he thinks about this from a design consideration. Does it fix a fatal error in php 7.2+? If so, we should probably merge and test it while waiting for his response.

The WordPress Core does not contain `session_start` anywhere in the codebase. Furthermore, treatment of `$_SESSION` within Core is always handled as if it were possible that it is not set (see wp-includes/load.php, which declares `$input = array_merge( $_GET, $_POST, $_COOKIE, $_SERVER, $_ENV, $_FILES, isset( $_SESSION ) && is_array( $_SESSION ) ? $_SESSION : array() );`). For this reason I don't think we should force it upon users, as there are performance and other implications, such as problems with headers being sent.

Can we use an ordinary cookie for persistence instead? @tombenner do you have thoughts on this?

**Comment by cyberscribe** [comment]
Also, how does WP Core handle flash-type messages that have to persist after the POST redirect? If we can follow their pattern, we can be minimally intrusive.

File: core/controllers/mvc_controller.php (hunk 1 above)
Comment (cyberscribe) [inline core/controllers/mvc_controller.php]: What does WP Core do to persist a message after the redirect from a POST in the admin? Best would be to follow their pattern.

**Comment by cyberscribe** [comment]
Will merge this and modify

**Pull request merged**